### PR TITLE
[PROF-11035] Add Ruby 3.4.0-rc1 headers

### PR DESCRIPTION
**Pull request by ivoanjo**
**What does this PR do?**

This PR adds the headers from the recently-released 3.4.0-rc1 release and also deletes the headers from 3.4.0-preview1.

**Motivation:**

There was a change to the layout of the `rb_thread_t *` structure after -preview2 that breaks the profiler's ability of reading the thread name.

I've locally tested and these headers fix the issue, and hopefully will turn out to be 100% compatible with the 3.4.0 stable release, giving us good support from day 0.

I've also chosen to delete 3.4.0-preview1 because it's no longer needed, as our CI is using preview2 or later. This also means that the total size of the gem should roughly stay the same, rather than growing.

**Additional Notes:**

I've also submitted an upstream PR to `debase-ruby_core_source`: https://github.com/ruby-debug/debase-ruby_core_source/pull/14 as I was doing the work anyway.

**How to test the change?**

Install ruby-3.4.0-rc1. Point dd-trace-rb's `Gemfile` to this PR. Validate that all profiler unit tests are green.

### CODE REVIEW

**Review by anmarchenko — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-# A sample Gemfile
 source "https://rubygems.org"
 
 gemspec
+
+gem 'pry'
```
Comment (lloeki) [inline Gemfile]: Could this be put in an optional group?

```suggestion
group :development, optional: true do
  gem 'pry'
end
```

**Review by lloeki — APPROVED** [review]
As I understand

- 3.4.0-preview1 is removed
- 3.4.0-preview2 is kept
- 3.4.0-rc1 is added

So it's all good with images that have a Ruby returning `3.4.0-preview2` as `RUBY_VERSION`?

If you would be so kind as to remove the empty commit from the history that would be nice.

File: Gemfile (hunk 1 above)
Comment (ivoanjo) [reply Gemfile]: Added in https://github.com/DataDog/datadog-ruby_core_source/pull/8/commits/c3e6c4c9a3b77eeeae58aed318fc96ccfd4a97cd

**Comment by ivoanjo** [comment]
> So it's all good with images that have a Ruby returning 3.4.0-preview2 as RUBY_VERSION?

This was actually my intention; yet on re-testing it looks like the matching logic we inherited from debase doesn't work very well.

Regardless of me trying to build on -rc1, -preview2 or even -preview1, it always picks the -rc1 headers. This doesn't seem new; I checked with the previous version of the gem that shipped -preview1 and -preview2, and compiling on -preview1 still picks the -preview2 headers -- it's just that nobody ever noticed because they were actually compatible, whereas the preview headers are not compatible with rc headers and vice-versa.

Let me see if I can include a little tweak to fix this...

**Comment by ivoanjo** [comment]
> Let me see if I can include a little tweak to fix this...

Actually, since is not new/related to this PR, I'll merge this PR and open a separate one to tackle this issue.

**Comment by ivoanjo** [comment]
Fix for version matching is in #11

**Pull request merged**